src/realm/object-store/c_api/sync.cpp

```diff
@@ -223,6 +223,19 @@ static realm_sync_error_code_t to_capi(const std::error_code& error_code, std::s
     return ret;
 }
 
+RLM_API char* realm_sync_client_get_path(realm_sync_error_t* sync_user_error) noexcept
```

Comment:
Is this really necessary? I imagine the SDK's can iterate through the user data themselves if the C API exposes the constants `c_original_file_path_key` and `c_recovery_file_path_key`?

If you want to keep this as a convenience to the API, you will also need to have a similar function for the recovery path. I would also suggest that you not duplicate strings here and just return a char pointer to the underlying data in the error struct so that the SDK doesn't have to worry about freeing these.

Reply:
I think it is better to expose the bare minimum, and the 2 constants are fine to be honest. Since we already do that for some other constant `string` part of `realm_sync_error_t`